lib/trino-hive-formats/src/main/java/io/trino/hive/formats/avro/ZstdAirCompressorCodec.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hive.formats.avro;
+
+import io.airlift.compress.v3.zstd.ZstdCompressor;
+import io.airlift.compress.v3.zstd.ZstdDecompressor;
+import io.airlift.compress.v3.zstd.ZstdInputStream;
+import org.apache.avro.file.Codec;
+import org.apache.avro.file.CodecFactory;
+import org.apache.avro.file.DataFileConstants;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+
+import static java.lang.Math.toIntExact;
+
+public class ZstdAirCompressorCodec
+        extends Codec
+{
+    public static final CodecFactory ZSTD_CODEC_FACTORY = new CodecFactory()
+    {
+        @Override
+        protected Codec createInstance()
+        {
+            return new ZstdAirCompressorCodec();
+        }
+    };
+
+    private final ZstdCompressor compressor = ZstdCompressor.create();
+    private final ZstdDecompressor decompressor = ZstdDecompressor.create();
+
+    @Override
+    public String getName()
+    {
+        return DataFileConstants.ZSTANDARD_CODEC;
+    }
+
+    @Override
+    public ByteBuffer compress(ByteBuffer uncompressedData)
+            throws IOException
+    {
+        if (uncompressedData.isDirect()) {
+            throw new IllegalArgumentException("Direct byte buffer not supported");
+        }
+        ByteBuffer output = ByteBuffer.allocate(compressor.maxCompressedLength(uncompressedData.remaining()));
+        int compressedSize = compressor.compress(
+                uncompressedData.array(), uncompressedData.arrayOffset() + uncompressedData.position(), uncompressedData.remaining(),
```

Comment:
nit: each argument in separate line

Reply:
I have the 3 inputs and the 3 outputs on their own line. Do you want all 6 on own line?